saleor/payment/models.py

```diff
@@ -0,0 +1,176 @@
+from decimal import Decimal
+from operator import attrgetter
+
+from django.conf import settings
+from django.contrib.postgres.fields import JSONField
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db import models
+from django_prices.models import MoneyField
+from prices import Money
+
+from . import (
+    ChargeStatus, CustomPaymentChoices, TransactionError, TransactionKind)
+from ..checkout.models import Cart
+from ..core.utils.taxes import zero_money
+from ..order.models import Order
+
+
+class Payment(models.Model):
+    """Represents transactable payment information
+    such as credit card details, gift card information or a customer's
+    authorization to charge their PayPal account.
+
+    All payment process related pieces of information are stored
+    at the gateway level, we are operating on the reusable token
+    which is a unique identifier of the customer for given gateway.
+
+    Several payment methods can be used within a single order.
+    """
+    # FIXME we should provide an option to store the card for later usage
+    # FIXME probably we should have pending status for 3d secure
+    gateway = models.CharField(max_length=255)
+    is_active = models.BooleanField(default=True)
+    #: Creation date and time
+    created = models.DateTimeField(auto_now_add=True)
+    #: Date and time of last modification
+    modified = models.DateTimeField(auto_now=True)
+    charge_status = models.CharField(
+        max_length=15,
+        choices=ChargeStatus.CHOICES,
+        default=ChargeStatus.NOT_CHARGED)
+
+    billing_first_name = models.CharField(max_length=256, blank=True)
+    billing_last_name = models.CharField(max_length=256, blank=True)
+    billing_company_name = models.CharField(max_length=256, blank=True)
+    billing_address_1 = models.CharField(max_length=256, blank=True)
+    billing_address_2 = models.CharField(max_length=256, blank=True)
+    billing_city = models.CharField(max_length=256, blank=True)
+    billing_city_area = models.CharField(max_length=128, blank=True)
+    billing_postal_code = models.CharField(max_length=256, blank=True)
+    billing_country_code = models.CharField(max_length=2, blank=True)
+    billing_country_area = models.CharField(max_length=256, blank=True)
+    billing_email = models.EmailField(blank=True)
+    customer_ip_address = models.GenericIPAddressField(blank=True, null=True)
```

Comment:
Wouldn't we need the billing phone number as well? It's useful for some gateways.

Reply:
Was wondering if that's not too much in terms of GDPR, is phone required for the process?

Reply:
In my the case of the payment gateway I'm targeting, it's to pass the phone number to the client's browser (toggle available in `settings.py`), with the user required to confirm they want to send their phone number shown on the screen to the advertised gateway.

It's not needed for the process, but required that user provides to finalize the process (client side).

I'm not sure how much it affects the GDPR, since we already have the phone number in the order address (which is an optional field), I think it could be acceptable as we don't give it away and it's not a nominative information. I may be really wrong, I have really limited knowledge on the subject.